lib/asciidoctor/stylesheets.rb

```diff
@@ -0,0 +1,106 @@
+# encoding: UTF-8
+# Monkey patch the whole file because the class is static
```

Comment:
One way to do it is to create the following file:

*lib/asciidoctor/ext/stylesheets.rb*
```ruby
module Asciidoctor
class Stylesheets
  def primary_stylesheet_data
    @primary_stylesheet_data ||= 'foobar'
  end 
end
end
```

Then require this path in *lib/asciidoctor/opal_ext.rb* (hijacking our opportunity to require additional scripts in core).

```ruby
module Asciidoctor; DATA_PATH = ''; end
require 'asciidoctor/stylesheets'
require 'asciidoctor/ext/stylesheets'
```

The challenge is that opal_ext gets required before the Asciidoctor module is defined, so we have to predefine the constant. If you have a way to add an extra require after requiring asciidoctor, use that instead.

The key point is that a monkeypatch needs to augment, not replace, the thing it's monkeypatching.

Of course, another solution to the problem is to literally patch the JS file as we do for the gsub stuff. The could be a cleaner approach.